src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -523,12 +529,14 @@ internal static bool TryGetMinimalCSharpRange(RazorCodeDocument codeDocument, Ra
 
             if (csharpResponse is null)
             {
-                // C# isn't ready yet, don't make Razor wait for it
+                // C# isn't ready yet, don't make Razor wait for it. Once C# is ready they'll send a refresh notification.
                 return SemanticTokensResponse.Default;
             }
             else if (csharpResponse.HostDocumentSyncVersion != null && csharpResponse.HostDocumentSyncVersion != documentVersion)
             {
-                // No C# response or C# is out of sync with us. Unrecoverable, return null to indicate no change. It will retry in a bit.
+                // No C# response or C# is out of sync with us. Unrecoverable, return null to indicate no change.
+                // Queue up a refresh notification to ask the client to request us again in a bit.
+                _semanticTokensRefreshPublisher.EnqueueWorkspaceSemanticTokensRefresh();
```

Comment:
In practice how often does this specific call trigger?

Reply:
In the "out of sync with us" case C# should already be sending a refresh once it syncs.

Reply:
Oh interesting, I'll delete this change then, thanks!